src/hidra/sender/eventdetectors/watchdog_events.py

```diff
@@ -463,8 +465,20 @@ def check_last_modified(self, filepath):
         global _events_marked_to_remove   # pylint: disable=invalid-name
 
         try:
-            # check modification time
             time_last_modified = os.stat(filepath).st_mtime
+            
+            # check modification time
+            # This modification is introduced to take care of the 0kb size files.
+            # 0kb size files will be monitored for 10 minutes and if no modification
+            # will happen then they will be taken out from potential_close_events
+            
+            minutes_diff = ( datetime.datetime.now() - datetime.datetime.fromtimestamp(time_last_modified)).total_seconds() / 60.0
+            file_size_check = os.stat(filepath).st_size
+            
+            if(minutes_diff > 10 and file_size_check == 0):
+                print("Due to the excessive wait in writting", filepath, "has been taken out of queue")
+                _events_marked_to_remove.append(filepath)
```

Comment:
Don't just remove the entry but create an event (it is not the purpose of this module to decide if empty files should be transferred but only to create events for all files matching the filters). This means we assume that after 10*`time_till_closed` without modification the file is probably closed despite having only 0 bytes.

Reply:
I think if a file is getting written then we should wait no matter how long. Until it is closed properly as the data is important. What my method is doing here is fairly simple. It will not consider zero byte files important after 10 minutes either they are closed or open.